code/modules/robotics/bot/chefbot.dm

```diff
@@ -39,117 +41,297 @@
 /obj/machinery/bot/chefbot/proc/why_is_it_bad()
 	return pick("IS FUCKING [pick("RAW", "BLAND", "UNDERCOOKED", "OVERCOOKED", "INEDIBLE", "RANCID", "DISGUSTING")]", "LOOKS LIKE [pick("BABY VOMIT", "A MUSHY PIG'S ASS", "REGURGITATED DONKEY SHIT", "A PILE OF ROTTING FLIES", "REFINED CAT PISS")]")
 
+#define FOOD_QUALITY_HORSESHIT 0
+#define FOOD_QUALITY_SHIT 1
+#define FOOD_QUALITY_GOOD_SHIT 2
 /obj/machinery/bot/chefbot/proc/yell()
 	if (!src.emagged || prob(50))
-		var/obj/item/reagent_containers/food/snacks/shitfood
+		var/obj/item/reagent_containers/food/snacks/food_to_judge
 		var/mob/living/carbon/human/thechef
 		var/mob/dork
 		var/is_thechef_the_chef = 0
+		var/how_shit = FOOD_QUALITY_HORSESHIT //ITS ALL DONKEY PISS UNTIL I SAY OTHERWISE
 		for (var/obj/item/reagent_containers/food/snacks/probablyshitfood in view(7, src))
-			if (probablyshitfood in calledout)
+			if (probablyshitfood.is_judged)
 				continue
```

Comment:
In a world with multiple Chefbots why wouldn't I be berated for my poor cooking multiple times?

Reply:
In my head every chefbot uses the same judging logic and as such a food item only needs to be judged once. If i instead decide to give each bot it's list of food items it has judged to remember then i end up with the issue Zewaka noted in their review.
Is there a more efficient alternative for food-list keeping?

Reply:
Could have an infinite cooldown with with something like `"judge_\ref[Object]"`, or could do something where it was a sufficiently long cooldown to justify it might think it is a different dish.  And then an easter egg where if the cooldown was part way saying that they judged it and now it is just stale/old/aged.

Reply:
Done, i now use an infinite cooldown whenever the bot judges a food. When looking for food items, if it has already been judged and passes a 5% chance check the chefbot yells about it being stale.